src/lucky/file_response.cr

```diff
@@ -0,0 +1,81 @@
+class Lucky::FileResponse < Lucky::Response
+  DEFAULT_STATUS = 200
+
+  getter context, path, disposition, filename, debug_message
+
+  def initialize(@context : HTTP::Server::Context,
+                 @path : String,
+                 @content_type : String? = nil,
+                 @disposition : String = "attachment",
+                 @filename : String? = nil,
+                 @status : Int32? = nil,
+                 @debug_message : String? = nil)
+  end
+
+  def print
+    raise Lucky::Exceptions::MissingFile.new("Cannot read file #{path}") unless file_exists?
+
+    set_response_headers
+    context.response.status_code = status
+    File.open(full_path) { |file| IO.copy(file, context.response) }
+  end
+
+  def status
+    @status || context.response.status_code || DEFAULT_STATUS
+  end
+
+  def set_response_headers
+    context.response.content_length = File.size(full_path)
+    context.response.content_type = content_type
+    context.response.headers["Accept-Ranges"] = "bytes"
+    context.response.headers["X-Content-Type-Options"] = "nosniff"
+    context.response.headers["Content-Transfer-Encoding"] = "binary"
+
+    if filename
```

Comment:
Just to make this extra clear, what do you think about something like `if filename_overridden?` or `if custom_filename?`?

Reply:
I like that, yeah. I think "custom_filename?" is a good one